src/vm/dllimportcallback.cpp

```diff
@@ -1111,28 +1111,32 @@ UMEntryThunk* UMEntryThunk::CreateUMEntryThunk()
 
     UMEntryThunk * p;
 
-#ifdef FEATURE_WINDOWSPHONE
     // On the phone, use loader heap to save memory commit of regular executable heap
     p = (UMEntryThunk *)(void *)SystemDomain::GetGlobalLoaderAllocator()->GetExecutableHeap()->AllocMem(S_SIZE_T(sizeof(UMEntryThunk)));
-#else
-    p = new (executable) UMEntryThunk;
-    memset (p, 0, sizeof(*p));
-#endif
 
     RETURN p;
 }
 
+#if defined(_TARGET_X86_) || defined(_TARGET_AMD64_)
+#define REDZONE_VALUE 0xcc
```

Comment:
We may introduce ``Spoil`` method as you suggested, which allows us to insert a breakpoint to the exact entry point (instead of filling the whole region).